src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -193,7 +197,7 @@ def resize_subvolume(self, **kwargs):
         try:
             with open_volume(self, volname) as fs_handle:
                 with open_group(fs_handle, self.volspec, groupname) as group:
-                    with open_subvol(fs_handle, self.volspec, group, subvolname) as subvolume:
+                    with open_subvol(fs_handle, self.volspec, group, subvolname, need_incarnation=True) as subvolume:
```

Comment:
Operations needing an incarnation are flagged of using the "need_incarnation" flag as in here.

Reply:
Current implementation uses the op_type and lets the subvolume version decide allowed states and types.